docs/design-proposals/2020-09-27-tidb-monitor-refactor.md

```diff
@@ -0,0 +1,56 @@
+# TiDB Monitor Refactor
+
+## Summary
+
+This document presents a design to refactor tidb monitor,solve some problems in monitoring uniformly.
+
+## Motivation
+
+Currently, it is not friendly that tidbmonitor monitor multiple cluster under the same namespaces or different namespaces.In a multi-cluster TLS environment, the cert of tidbmonitor is bound to the first cluster of the `cluster` Spec ,we should eliminate this binding relationship.
+Secondly,tidbmonitor is also a statefulset application,we should switch to statefulset deploy from deployment crd,this process should smooth upgrade.
+Also, many users will integrate the company’s existing monitoring system, we should support thanos sidecar spec refer to the Prometheus operator, and optimize metric service and provide googe yaml example like `ServiceMonitor` ,`thanos` etc.
+Others are some small optimization points,providing more field for `kubectl get tidbmonitor`  command, supporting multiple cluster grafana dashborad,optimizing pd dashboard address writing logic,fulling e2e and unit testing etc.
+
+After discussion, we will not support monitor multiple clusters in tls environments because users must ensure that all monitored clusters use the same CA certificate,but autos-scale cluster or heterogeneous cluster need to use origin cluster certificate.
+We want to split the client secret of the monitor and the client secret of the cluster,eliminate this binding relationship that tidbmonitor secret bound to the firsrt cluster of the `cluster` Spec in mutiple clusters.But the user must rely on himself to ensure that the monitor's certificate is correct.
+We think it is unreliable rely on user himself,so we just support monitor multiple clusters in non-tls environment and providing other methods to solve the monitoring of tls environment,for example thanos query. 
+
+### Goals
+
+* Make TidbMonitor able to monitor multiple TidbClusters at non-TLS environment.
+* Improve the monitor's Unit and E2E test.
+* Smooth upgrade deployment to statefulset.
+* Support thanos sidecar and more friendly integration of Prometheus operator.
+
+### Non-Goals
+
+
+## Proposal
```

Comment:
We'd better add some high-level descriptions for each item to describe the solution.

Reply:
ok

Reply:
* Improve the robustness of tidbmonitor,for example multiple cluster,code level.
* Meet most of the needs of users and make monitoring easier to use.
Is it ok?

Reply:
I mean for the following items, we need to add some detail about what we should do.